dotnet-authserver/src/TeacherIdentity.AuthServer/Services/DqtApi/TeacherNameChangeRequest.cs

```diff
@@ -0,0 +1,11 @@
+namespace TeacherIdentity.AuthServer.Services.DqtApi;
+
+public record TeacherNameChangeRequest
+{
+    public required string FirstName { get; init; }
+    public string? MiddleName { get; init; }
```

Comment:
Even though it's nullable I think forcing this to be explicitly set is no bad thing
```suggestion
    public required string? MiddleName { get; init; }
```